### PR TITLE
fix warning on php 8

### PR DESCRIPTION
**Pull request by lazyants**
Only the first byte will be assigned to the string offset at /var/www/vendor/mikehaertl/php-pdftk/src/XfdfFile.php:82

### CODE REVIEW

File: src/XfdfFile.php [hunk 1]
```diff
@@ -77,7 +77,12 @@ public function __construct($data, $suffix = null, $prefix = null, $directory =
                     }
                     $target = & $target[$part];
                 }
-                $target[$final] = $sanitizedValue;
+                
+                if (is_string($target)) {
+                    $target = $sanitizedValue;
+                } else {
+                    $target[$final] = $sanitizedValue;
+                }
```
Comment (mikehaertl) [inline src/XfdfFile.php:85]: @lazyants This class was not written by me and to be honest I had to read this code block above your change a couple of times to get what it does. The references make it hard to follow. 

This whole block really needs a comment with an example how the final array will look like. As you probably spent some more time on this than me, maybe you could add something? And maybe you could also give an example why your change is required here? I would have said that `$target` can't be a string here anyway - but that's probably not correct.

Oh and if you find time - a test that shows that your fix works would also be great.

Comment (lazyants) [reply src/XfdfFile.php:85]: @mikehaertl 
normally it works as it is. The problem occurs only if u use array as input values.

So, this will work well:

```
 "delivery-checkbox0" => "O"
 "delivery-share-avg0" => "2,4321"
 "delivery-checkbox1" => "N"
 "delivery-share-avg1" => "3,1234"
```

So, this one will produce warnings in php8 (but will still work if you mute the warning even without my pr):

```
 "delivery-checkbox0" => "O"
 "delivery-share-avg0" => "2,4321"
 "delivery-checkbox1" => "N"
 "delivery-share-avg1" => "3,1234"
```

Unfortunately I currently don't have time to write proper tests because there are no for this class which I could take as a kind draft, but probably you can merge the pr without tests for now.

Comment (mikehaertl) [reply src/XfdfFile.php:85]: @lazyants There *are* tests and we even test the dot format there:
https://github.com/mikehaertl/php-pdftk/blob/master/tests/XfdfFileTest.php

But no worries. I'll have a look why the tests don't fail currently. Maybe warnings are disabled - which they shouldn't be.

Comment (mikehaertl) [reply src/XfdfFile.php:85]: @lazyants I'm sorry, but I can't really reproduce the issue. I also tried the examples you gave and they work fine. I can't see how `$target` could be a string at this point. It initially is set to `$fields` which is an array. And then with each new nesting level is initiated to an empty array in line 76. 

The only way I could make it fail is with invalid data like this:

```
$data = [
    'address' => 'bla',
    'address.name' => 'other bla',
];
```

But this is not valid because `address` can either be a string value or be the first part of a hierarchical field in dot notation but not both.

Could you maybe show the data that you used?

Comment (lazyants) [reply src/XfdfFile.php:85]: So now it found as well. 
the problem occurs if you use 

```php
$data = [
    'item' => 'bla',
    'item.1' => 'other bla',
];
```

but not if you use

```php
$data = [
    'item.0' => 'bla',
    'item.1' => 'other bla',
];
```

And on one side you are right with string and array, on the other side it has been worked under php 7.x and works also under 8.x (apart of warning), so probably it makes sense to add the check from above?

Comment (mikehaertl) [reply src/XfdfFile.php:85]: The thing is that this data:

```php
$data = [
    'item' => 'bla',
    'item.1' => 'other bla',
];
```

would generate this XML in XFDF:

```xml
<field name="item">
  <value>bla</value>
  <field name="1">
    <value>other bla</value>
  </field>
</field>
```

But that's not a valid form. There can't be a `<value>` *and* another `<field>` inside a `<field>`. So I would argue, that the error indicates that your data is malformed and needs to be fixed.

Without the warning you would never find this issue.

Comment (mikehaertl) [reply src/XfdfFile.php:85]: One more comment here: It worked before just by accident, because in the first iteration the final array looks like this:

```
[
    'item' => 'bla',
],
```

while in the next iteration the `item` key value is overwritten:

```
[
    'item' => [
        '0' => 'other bla',
    ],
]
```

So your inital value `bla` for `item` is lost and will never be used - without any further notice. I still think that it's OK that PHP throws a warning here. It means that your code does something unexpected and needs your attention.

One could argue that we could throw an exception - but this would really be a BC break.

So unless you totally disagree I would close this PR.

Comment (lazyants) [reply src/XfdfFile.php:85]: ok, interesting, thank you, I'll check!

**Comment by mikehaertl** [comment]
Closing this, as discussed.
